hir/src/function.rs

```diff
@@ -224,6 +224,15 @@ impl PartialEq for Signature {
     }
 }
 
+/// Represents the lifted(component) type of a component imported/exported function
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LiftedFunctionType {
```

Comment:
**Nit:** I would place this in the `miden-hir-type` crate, since it mirrors `FunctionType` which is defined in that crate. All types in that crate are exported from the `miden-hir` root, so it shouldn't make a difference for any downstream crates.

Reply:
Sure. Moved.